pkg/components/ebpf/common/httpfltr_transform.go

```diff
@@ -75,15 +76,29 @@ func ReadHTTPInfoIntoSpan(record *ringbuf.Record, filter ServiceFilter) (request
 		return request.Span{}, true, nil
 	}
 
-	return HTTPInfoEventToSpan(event)
+	return HTTPInfoEventToSpan(parseCtx, event)
 }
 
-func HTTPInfoEventToSpan(event *BPFHTTPInfo) (request.Span, bool, error) {
+func HTTPInfoEventToSpan(parseCtx *EBPFParseContext, event *BPFHTTPInfo) (request.Span, bool, error) {
 	result := HTTPInfo{BPFHTTPInfo: *event}
 
-	var bufHost string
-	var bufPort int
-	parsedHost := false
+	var (
+		bufHost       string
+		bufPort       int
+		parsedHost    bool
+		requestBuffer []byte
+	)
+
+	if event.HasLargeBuffers == 1 {
+		b, ok := extractTCPLargeBuffer(parseCtx, event.Tp.TraceId, event.Tp.SpanId, packetTypeRequest)
```

Comment:
Actually, can we please change this to fallback to the event.Buf[:] if we fail to extract the buffer? No need to reject the span. I am thinking this is the reason the integration tests are failing.

Reply:
Yes I can do that, I am not sure the failing test is related though, the same test just failed in my other PR -> https://github.com/open-telemetry/opentelemetry-ebpf-instrumentation/actions/runs/18014863416/job/51257072591 😞

Reply:
Yeah possibly not, it passed on the latest re-run, I'll take a look what's up. Maybe just make the safe fix anyway so that we get the most spans, better something than nothing.